tests/parser/LgrTests.cpp

```diff
@@ -101,6 +101,7 @@ SCHEDULE
     Opm::EclipseState state(deck);
     Opm::LgrCollection lgrs = state.getLgrs();
 
+    BOOST_CHECK(state.hasInputLGR());
```

Comment:
Would you mind using `BOOST_CHECK_MESSAGE()` for this?  A readable diagnostic message with more context is usually preferable to just "Test X failed".